usaspending_api/search/v2/views/spending_by_category_views/base_spending_by_location.py

```diff
@@ -0,0 +1,119 @@
+from abc import ABCMeta
+from enum import Enum
+from typing import List
+
+from django.db.models import QuerySet, Sum
+from elasticsearch_dsl import A, Q as ES_Q
+
+from usaspending_api.common.elasticsearch.search_wrappers import TransactionSearch
+from usaspending_api.common.helpers.api_helper import alias_response
+from usaspending_api.search.helpers.spending_by_category_helpers import (
+    fetch_country_name_from_code,
+    fetch_state_name_from_code,
+)
+from usaspending_api.search.v2.elasticsearch_helper import get_sum_aggregations
+from usaspending_api.search.v2.views.spending_by_category import ALIAS_DICT
+from usaspending_api.search.v2.views.spending_by_category_views.base_spending_by_category import (
+    BaseSpendingByCategoryViewSet,
+)
+
+
+class LocationType(Enum):
+    COUNTY = "county"
+    DISTRICT = "congressional"
+    COUNTRY = "country"
+    STATE = "state"
+
+
+class BaseLocationViewSet(BaseSpendingByCategoryViewSet, metaclass=ABCMeta):
+    """
+    Base class used by the different Awarding / Funding Agencies and Subagencies
```

Comment:
Short enough to be a single-line docstring

Reply:
fixed